src/systemd/sd-journal.h

```diff
@@ -72,6 +72,7 @@ enum {
         SD_JOURNAL_ALL_NAMESPACES            = 1 << 5, /* Show all namespaces, not just the default or specified one */
         SD_JOURNAL_INCLUDE_DEFAULT_NAMESPACE = 1 << 6, /* Show default namespace in addition to specified one */
         SD_JOURNAL_TAKE_DIRECTORY_FD         = 1 << 7, /* sd_journal_open_directory_fd() will take ownership of the provided file descriptor. */
+        SD_JOURNAL_READ_TAIL_TIMESTAMP_ONCE  = 1 << 8, /* Read timestamp of tail entry only when journal files are opened. */
```

Comment:
please give this a more generic name, i.e. abstract the optimization this allows internally.

i.e. call this SD_JOURNAL_ASSUME_IMMUTABLE or so.

This leaves the door open to hide other stuff behind this. Also if the flag is set, make sure that sd_journal_get_fd(), sd_journal_get_events(), sd_journal_get_timeout(), sd_journal_process(), sd_journal_wait() behave differently. They could either return some recognizable error if you try to use them with the flag. Or they could return "dummy" stuff, i.e. indicate that there's no new input, whatever you do.

But I think we should probably return a recognizable in this case, simply to ensure misuses are clearly recognizable. i.e. return -EUNATCH or so in this case?